cliquet/authorization.py

```diff
@@ -19,6 +19,18 @@ class AuthorizationPolicy(object):
     get_bound_permissions = None
 
     def permits(self, context, principals, permission):
+        # Add prefixed user id to principals.
+        if context.prefixed_userid:
+            principals = principals + [context.prefixed_userid]
+            prefix, user_id = context.prefixed_userid.split(':', 1)
+            # Remove unprefixed user id to avoid conflicts.
+            # (it is added via Pyramid Authn policy effective principals)
+            if user_id in principals:
+                principals.remove(user_id)
+            # Retro-compatibility with cliquet 2.0 '_' user id prefixes.
+            # Just in case it was used in permissions definitions.
+            principals.append('%s_%s' % (prefix, user_id))
```

Comment:
Do we really want to handle retrocompatibility here since we don't have anything in production yet?